src/huggingface_hub/hf_api.py

```diff
@@ -644,7 +656,15 @@ def list_models(
         r = requests.get(path, params=params, headers=headers)
         r.raise_for_status()
         d = r.json()
-        return [ModelInfo(**x) for x in d]
+        res = [ModelInfo(**x) for x in d]
+        if emissions_thresholds is not None:
+            if cardData is None:
+                raise ValueError(
+                    "`emissions_thresholds` were passed without setting `cardData=True`."
+                )
+            else:
+                return _filter_emissions(res, *emissions_thresholds)
```

Comment:
Should this return the model indices of the models? Or should it return the list results? (I feel that we want the list results, since this is essentially a filter).

Reply:
My vote is the list results for similar reasons